compiler/backend/src/org/jetbrains/kotlin/codegen/inline/AnonymousObjectTransformer.kt

```diff
@@ -113,21 +114,14 @@ class AnonymousObjectTransformer(
             override fun visitEnd() {}
         }, ClassReader.SKIP_FRAMES)
 
-        if (!inliningContext.isInliningLambda) {
-            sourceMapper = if (debugInfo != null && !debugInfo!!.isEmpty()) {
-                SourceMapper.createFromSmap(SMAPParser.parse(debugInfo!!))
-            } else {
-                //seems we can't do any clever mapping cause we don't know any about original class name
-                IdenticalSourceMapper
-            }
-            if (sourceInfo != null && !GENERATE_SMAP) {
-                classBuilder.visitSource(sourceInfo!!, debugInfo)
-            }
+        // When regenerating objects in lambdas, we have to pass the SMAP straight through and keep
+        // the original line numbers because SMAPParser does not parse call site markers.
```

Comment:
This also means call sites are currently lost during regeneration:

```
fun box() {
    foo<Any>()
}

inline fun <reified T> foo() = object {
    fun run() = bar<T>()
}.run()

inline fun <reified T> bar() {
    null as T
}

// NAVIGATE_TO_CALL_SITE
// FILE: inlineFunCallSiteInRegeneratedObject.kt
// LINE: 6
```
There's no option to navigate to `bar`'s call site.

Reply:
Is it possible to parse them and make some wise remapping if needed (maybe with additional support in navigation) to support `bar` navigation? //cc @yanex

Reply:
Of course, just might take some time to ensure the parser is as general as possible.

Reply:
So I wrote the code for that, but then realized it's not that easy because in this case call site markers might need to point to several files. E.g. the KotlinDebug stratum of the regenerated object in
```
// FILE: 1.kt
fun box() {
    foo {
        bar()
    }
}
// FILE: 2.kt
inline fun foo(crossinline x: () -> Unit) = object {
    fun run() {
        bar()
        x()
    }
}.run()

inline fun <reified T> bar() = ...
```
should map some line numbers to 2.kt:4 and some to 1.kt:4. This requires *even more* refactoring of SourceMapper and SMAPBuilder.

Reply:
Done.